### PR TITLE
test(bot): integration coverage for blendAutoplayTracks (#1089)

### LINKED ISSUE #1089: test(play): integration test for blendAutoplayTracks queue mutation
## Context
`blendAutoplayTracks` mutates `queue.tracks` and triggers a background replenish, but the `/play` spec only asserts the mock was *called* — it never verifies the queue was actually mutated correctly or that replenish fired. This leaves the core blend behavior untested.

## Evidence
- `packages/bot/src/functions/music/commands/play/handlers/playHandler.ts:233` — calls `blendAutoplayTracks`
- `play/index.spec.ts` mocks `blendAutoplayTracks` (delegation assertion only)

## Approach
1. Add an integration-style test: seed a queue with N autoplay tracks, call `blendAutoplayTracks(ratio)`, assert the expected number removed and replenish invoked.
2. Cover edge cases: empty autoplay pool, ratio 0 and 1.

## Acceptance criteria
- Test asserts actual `queue.tracks` mutation, not just mock invocation
- Replenish trigger verified
- Edge cases (empty pool, ratio bounds) covered

---
<sub>Surfaced by `/backlog` (Play + autoplay scope) on 2026-05-28. Dedup key: `code:playHandler:233:blend-integration-test`.</sub>

### PR DESCRIPTION
**Pull request by LucasSantana-Dev**
## Summary
- Adds 6 new integration-level tests for `blendAutoplayTracks` function
- Exercises REAL implementation with actual queue.tracks mutations
- Verifies replenish queue invocation and error resilience

## Test Coverage
- **blendRatio 0.5:** Removes 50% of autoplay tracks (keepCount = ceil(4 * 0.5) = 2)
- **blendRatio 0:** Removes all autoplay tracks (empty autoplay pool)
- **blendRatio 1:** Keeps all autoplay tracks
- **Mixed queue:** Preserves user tracks while removing autoplay tracks
- **Replenish invocation:** Verifies replenishQueue called with correct queue
- **Error resilience:** Gracefully handles individual track removal failures

## Test Results
- **queueEditOps.spec.ts:** 28 passed
- **Full bot suite:** 2489 passed, 6 skipped

## Verification
```bash
cd packages/bot && npx jest src/utils/music/queueEditOps.spec.ts
# Result: Test Suites: 1 passed, Tests: 28 passed
```

Closes #1089

@cubic-dev-ai

---
## Summary by cubic
Add 6 integration tests for `blendAutoplayTracks` to validate real queue.tracks mutations and replenish behavior. Covers ratios 0/0.5/1, preserves user tracks in mixed queues, invokes replenish with the queue, and includes a remove-failure case that mutates the live queue.

<sup>Written for commit 028332c8ace316ca00c62ec2e9d5829879bfa49b. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/LucasSantana-Dev/Lucky/pull/1369?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/bot/src/utils/music/queueEditOps.spec.ts [hunk 1]
```diff
@@ -345,4 +345,165 @@ describe('blendAutoplayTracks', () => {
...
+            const auto3 = createTrack('Auto3', 'Bot', 'u3')
+            for (const t of [auto1, auto2, auto3]) {
+                ;(t as unknown as { metadata: Record<string, unknown> }).metadata =
+                    { isAutoplay: true }
+            }
+            const queue = createQueue([auto1, auto2, auto3])
+
+            // Make node.remove fail on the second call
+            const removeCallCount = { count: 0 }
+            ;(queue.node.remove as jest.Mock).mockImplementation(
```
[the remark this answers is not in the document]
Comment (LucasSantana-Dev) [reply packages/bot/src/utils/music/queueEditOps.spec.ts:484]: Valid — the override spliced a toArray() copy, so the size assertion passed vacuously. Fixed in 028332c8: the mock now delegates successful calls to the stub's live-array implementation, and the assertion tightened to size 2 (one real removal + one swallowed failure).
